go/service/identify.go

```diff
@@ -357,16 +357,18 @@ func (h *IdentifyHandler) resolveIdentifyImplicitTeamDoIdentifies(ctx context.Co
 			if idErr != nil {
 				h.G().Log.CDebugf(subctx, "Failed to convert result from Identify2: %s", idErr)
 			}
-			if idRes != nil && idRes.TrackBreaks != nil && idErr == nil {
-				trackBreaksLock.Lock()
-				defer trackBreaksLock.Unlock()
-				if res.TrackBreaks == nil {
-					res.TrackBreaks = make(map[keybase1.UserVersion]keybase1.IdentifyTrackBreaks)
+			if idRes != nil {
+				if idRes.TrackBreaks != nil && idErr == nil {
+					trackBreaksLock.Lock()
+					defer trackBreaksLock.Unlock()
+					if res.TrackBreaks == nil {
+						res.TrackBreaks = make(map[keybase1.UserVersion]keybase1.IdentifyTrackBreaks)
+					}
+					res.TrackBreaks[idRes.Upk.ToUserVersion()] = *idRes.TrackBreaks
+					brokenUsernames = append(brokenUsernames, idRes.Upk.GetName())
+				} else {
+					okUsernames = append(okUsernames, idRes.Upk.GetName())
```

Comment:
Looks ok, but does this need to be protected by a lock too?

Reply:
yes, good catch!